lib/cambiatus/workers/monthly_digest_worker.ex

```diff
@@ -1,23 +1,39 @@
 defmodule Cambiatus.Workers.MonthlyDigestWorker do
   @moduledoc """
-  Handles monthly news digest
+  Handles monthly news digest by creating workers to send emails
   """
   use Oban.Worker, queue: :monthly_digest
 
   alias Cambiatus.Accounts.User
   alias Cambiatus.Commune.Community
   alias Cambiatus.Repo
   alias Cambiatus.Social.News
+  alias Cambiatus.Workers.DigestEmailWorker
+  alias Ecto.Multi
 
   def perform(_) do
-    Community
-    |> Community.with_news_enabled()
-    |> Repo.all()
-    |> Repo.preload([[news: News.last_thirty_days()], [members: User.accept_digest()], :subdomain])
-    |> Enum.each(fn community ->
-      unless Enum.empty?(community.news) do
-        CambiatusWeb.Email.monthly_digest(community)
-      end
-    end)
+    multi = Multi.new()
+
+    transaction =
+      Community
+      |> Community.with_news_enabled()
+      |> Repo.all()
+      |> Repo.preload([
+        [news: News.last_thirty_days()],
+        [members: User.accept_digest()],
+        :subdomain
+      ])
+      |> Enum.filter(&Enum.any?(&1.news))
+      |> Enum.reduce(multi, fn community, multi ->
```

Comment:
I think this would be a bit clearer - we wouldn't have so many `multi` around:

```suggestion
      |> Enum.reduce(Multi.new(), fn community, multi ->
```

Reply:
More concise indeed